frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-ref.ts

```diff
@@ -22,6 +27,42 @@ export const getReference = ({
   kind,
 }: K8sGroupVersionKind): K8sResourceKindReference => [group || 'core', version, kind].join('~');
 
+/**
+ * @deprecated - This will become obsolete when we move away from K8sResourceKindReference to K8sGroupVersionKind
+ * Provides an object containing group and version for given apiVersion string.
+ * @param apiVersion apiVersion
+ * @return The object containing group and version for given apiVersion string.
+ * * */
+export const groupVersionFor = (apiVersion: string) => ({
+  group: apiVersion.split('/').length === 2 ? apiVersion.split('/')[0] : 'core',
+  version: apiVersion.split('/').length === 2 ? apiVersion.split('/')[1] : apiVersion,
+});
+
+// TODO(alecmerdler): Replace all manual string building with this function
+/**
+ * @deprecated - This will become obsolete when we move away from K8sResourceKindReference to K8sGroupVersionKind
+ * Provides a reference string that uniquely identifies given group, version, and kind.
+ * @param group group
+ * @param version version
+ * @param kind kind
+ * @return The reference for group, version and kind i.e `group~version~kind`.
+ * * */
+export const referenceForGroupVersionKind = (group: string, version: string, kind: string) =>
+  [group, version, kind].join('~');
+
+/**
+ * @deprecated - This will become obsolete when we move away from K8sResourceKindReference to K8sGroupVersionKind
+ * Provides a reference string that uniquely identifies the group, version, and kind of an owner reference.
+ * @param ownerRef owner reference
+ * @return The reference for owner reference i.e `group~version~kind`.
+ * * */
+export const referenceForOwnerRef = (ownerRef: OwnerReference): GroupVersionKind =>
+  referenceForGroupVersionKind(
+    groupVersionFor(ownerRef.apiVersion).group,
+    groupVersionFor(ownerRef.apiVersion).version,
+    ownerRef.kind,
+  );
```

Comment:
This feels like a means to an end... the `ResourceIcon` could just be enhanced to accept the K8sGroupVersionKind and internally to `ResourceIcon` we could call `getReference`. `modelFor` could just use `useK8sModel`. There are alternatives to clean up the references instead of just pulling everything into the SDK.

This utility function arguably is fine, but it shouldn't be returning a `GroupVersionKind`. I think going from an `OwnerReference` to an ID for the K8sModel (in this case `K8sGroupVersionKind`).